apps/uci/web/src/app/home/page.tsx

```diff
@@ -0,0 +1,231 @@
+import React, { useContext, useEffect, useMemo, useState } from 'react';
+import { Box, Flex, useBreakpointValue, Tabs, TabList, TabPanels, Tab, TabPanel, Text, Tooltip, IconButton, Heading, Input, InputGroup, InputRightElement, InputLeftElement } from '@chakra-ui/react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { filter, forEach } from 'lodash';
+import { useRouter } from 'next/navigation';
+import { toast } from 'react-hot-toast';
+import styles from './index.module.css';
+import ChatItem from '@/components/common/chat-item';
+import moment from 'moment';
+import StarredChatItem from '@/components/common/starred-chat-item';
+import { User } from '@/types';
+import { AppContext } from '@/context';
+import { config } from '@/config';
+
+export default function Home() {
+  const { currentUser, allUsers, setMessages } = useContext(AppContext);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  useEffect(() => {
+    try {
+      const userMsgsFromLocal = JSON.parse(localStorage.getItem('userMsgs') || '');
+      if (userMsgsFromLocal?.length > 0) {
+        const userMsgs = filter(userMsgsFromLocal, {
+          botUuid: currentUser?.id,
+        });
+        setMessages(userMsgs);
+      }
+    } catch (err: any) {
+      toast.error(err?.message || err);
+    }
+  }, [setMessages, currentUser?.id]);
+
+  const history = useRouter();
+  const context = useContext(AppContext);
+
+  // const onStarredChatsClick = useCallback((): void => {
+  //   history.push('/starred-chats');
+  // }, [history]);
+
+  useEffect(() => {
+    setTimeout(() => {
+      context?.setLoading(false);
+      if (document.getElementById('mainLoader')) {
+        document.getElementById('loader').setAttribute('display', 'none');
+        toast.error('चैटबॉट जवाब नहीं दे पा रहा हैं। कृपया बाद में पुन: प्रयास करें।');
+      }
+    }, 60000);
+  }, [context]);
+
+  const isMobile = useBreakpointValue({ base: true, md: false });
+
+  const sortedUsers = useMemo(() => {
```

Comment:
try to reuse the logic from getBotDetails part.
Use lodash sortBy,reverse and filter method for this